TS2880: legacy `assert { "resolution-mode": "import" }` in vite-plugin.d.ts breaks type-checking with TypeScript 6 / tsgo

## Problem

`vite-plugin.d.ts` (and its source `src/vite-plugin.ts`) uses the legacy import-assertion syntax:

```ts
import type { ResolvedConfig } from "vite" assert { "resolution-mode": "import" };
```

Import assertions were replaced by import attributes, and TypeScript 6 (as well as tsgo / `@typescript/native-preview`) now rejects the old `assert` keyword as a grammar error:

```
node_modules/vite-envs/vite-plugin.d.ts(1,44): error TS2880: Import assertions have been replaced by import attributes. Use 'with' instead of 'assert'.
```

Because TS2880 is a **syntax** error raised while parsing the declaration file, `skipLibCheck: true` does not suppress it — any project that imports `vite-envs` in its Vite config fails to type-check.

## Reproduction

- `vite-envs@4.7.1` (latest)
- `typescript@6.0.3` or `@typescript/native-preview` (tsgo)
- A `vite.config.ts` containing `import { viteEnvs } from "vite-envs"`, covered by a tsconfig
- Run `tsc -b` / `tsgo -b` → TS2880

## Suggested fix

Replace `assert` with `with` in `src/vite-plugin.ts` (and the published `vite-plugin.d.ts`):

```ts
import type { ResolvedConfig } from "vite" with { "resolution-mode": "import" };
```

The `with` syntax is supported since TypeScript 5.3, so this should not break consumers on older (≥5.3) TypeScript versions. I verified locally (via `pnpm patch`) that this one-line change fixes the error.

Thanks for the great package!

--- a/src/vite-plugin.ts
+++ b/src/vite-plugin.ts
@@ -7,7 +7,7 @@ import {
     relative as pathRelative,
     normalize as pathNormalize
 } from "path";
-import type { Plugin, ResolvedConfig } from "vite" assert { "resolution-mode": "import" };
+import type { Plugin, ResolvedConfig } from "vite" with { "resolution-mode": "import" };
 import { assert } from "tsafe/assert";
 import { typeGuard } from "tsafe/typeGuard";
 import { getThisCodebaseRootDirPath } from "./tools/getThisCodebaseRootDirPath";

